internal/snmp/translate.go

```diff
@@ -119,6 +119,13 @@ func TrapLookup(oid string) (e MibEntry, err error) {
 		return e, err
 	}
 
+	givenOid := strings.Split(oid, ".")
+	if givenOid[0] == "" {
+		givenOid = givenOid[1:]
+	}
+
+	foundOid := strings.Split(node.Oid.String(), ".")
```

Comment:
Oid is already an array, no need to concat and split again,
```suggestion
	givenOid := types.OidMustFromString(oid)
	foundOid := node.Oid
```

I think defining `foundOid` isn't needed in this case and `node.Oid` can be used directly later in that loop.

Also, `types.OidFromString` should be used, as the other one might panic.